src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -2,81 +2,70 @@
             }
             _existingState = existingState ?? throw new ArgumentNullException(nameof(existingState));
         }
 
         /// <summary>
-        /// Represents the method that performs operations when <see cref="OnPersisting"/> is raised and the application is about to be paused.
+        /// Register a callback to persist the component state when the application is about to be paused.
+        /// Registered callbacks can use this opportunity to persist their state so that it can be retrieved when the application resumes.
         /// </summary>
-        /// <returns>A <see cref="Task"/> that will complete when the method is done preparing for the application pause.</returns>
-        public delegate Task OnPersistingCallback();
-
-        /// <summary>
-        /// An event that is raised when the application is about to be paused.
-        /// Registered handlers can use this opportunity to persist their state so that it can be retrieved when the application resumes.
-        /// </summary>
-        public event OnPersistingCallback OnPersisting
+        /// <param name="callback">The callback to invoke when the application is being paused.</param>
+        /// <returns>A subscription that can be used to unregister the callback when disposed.</returns>
+        public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> callback)
         {
-            add
+            if (callback == null)
             {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
-
-                _registeredCallbacks.Add(value);
+                throw new ArgumentNullException(nameof(callback));
             }
-            remove
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
 
-                _registeredCallbacks.Remove(value);
-            }
+            _registeredCallbacks.Add(callback);
+
+            return new PersistingComponentStateSubscription(_registeredCallbacks, callback);
         }
 
         /// <summary>
         /// Tries to retrieve the persisted state with the given <paramref name="key"/>.
         /// When the key is present, the state is successfully returned via <paramref name="value"/>
-        /// and removed from the <see cref="ComponentApplicationState"/>.
+        /// and removed from the <see cref="PersistentComponentState"/>.
         /// </summary>
         /// <param name="key">The key used to persist the state.</param>
         /// <param name="value">The persisted state.</param>
         /// <returns><c>true</c> if the state was found; <c>false</c> otherwise.</returns>
-        public bool TryTakePersistedState(string key, [MaybeNullWhen(false)] out byte[]? value)
+        public bool TryTake(string key, [MaybeNullWhen(false)] out ReadOnlySequence<byte> value)
```

Comment:
```suggestion
        public bool TryTake(string key, [NotNullWhen(true)] out ReadOnlySequence<byte>? value)
```

Reply:
This was wrong all-together. ReadonlySequence is a struct.